scalyr_agent/builtin_monitors/syslog_monitor.py

```diff
@@ -1591,33 +1594,51 @@ def __handle_syslog_logs(self, data, extra):
                         limit_key="syslog-max-log-files",
                     )
                 else:
-                    logfile = AutoFlushingRotatingFile(
-                        filename=path,
-                        max_bytes=self.__max_log_size,
-                        backup_count=self.__max_log_rotations,
-                        flush_delay=self.__flush_delay,
-                    )
+                    # Match the new log path with a global log config (generally globbed).
+                    # Note that in log configs, .path is required but .attributes may be empty.
+                    log_config_attribs = None
+                    for log_config in self.__global_log_configs:
+                        if fnmatch(path, log_config["path"]):
```

Comment:
IIRC, we will only end up in this code path in log file for a specific path doesn't exist yet so it shouldn't add much overhead since it won't be called for every single syslog message / log line, right?

Reply:
Also, We should probably try to add more debug logs to this function to make troubleshooting easier.

As long as we add it to the code paths which only run for a new log file (and not for every single syslog message), it shouldn't add much overhead and should be fine to do it.

Reply:
Yes that's right.  Sure, added a couple debug logs.